### PR TITLE
[5G: MVC][AMF] AMF Set Interface Server

### PR DESCRIPTION
**Pull request by ronit-kumar-acl**
Signed-off-by: ronit-kumar-acl <ronit.k@altencalsoftlabs.com>

## Summary

Added AMF server implementation inside oai/tasks.
Receives Set interface grpc requests when client services use "amf_service"  as the name of the channel in the client code. 

## Test Plan

Unit Test logs: [amf_server_unit_test.log](https://github.com/magma/magma/files/5383827/amf_server_unit_test.log)
## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Review by ulaskozat — APPROVED** [review]
LGTM

File: lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.cpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <string>
+
+#include "lte/protos/session_manager.pb.h"
+#include "lte/protos/spgw_service.pb.h"
+#include <folly/IPAddress.h>
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.cpp]: are you using this in this PR?

File: lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.cpp [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <string>
+
+#include "lte/protos/session_manager.pb.h"
+#include "lte/protos/spgw_service.pb.h"
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.cpp]: are you using this in this PR?

File: lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.cpp [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+#include <folly/IPAddress.h>
+
+extern "C" {
+#include "spgw_service_handler.h"
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.cpp]: are you using this?

File: lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.h [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+limitations under the License.
+*/
+
+#pragma once
+
+#include <grpc++/grpc++.h>
+#include <grpcpp/impl/codegen/status.h>
+
+#include "lte/protos/session_manager.grpc.pb.h"
+#include "lte/protos/spgw_service.grpc.pb.h"
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.h:18]: are you using this?

File: lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.h [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+#include "lte/protos/policydb.pb.h"
+
+extern "C" {
+#include "spgw_service_handler.h"
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.h:22]: are you using this?

**Comment by ulaskozat** [comment]
Please address the review comments before any merging and mark them as resolved.

**Review by ulaskozat — CHANGES_REQUESTED** [review]
see the comments.

File: lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.cpp (hunk 1 above)
Comment (ronit-kumar-acl) [reply lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.cpp]: Yes,  session_manager.pb.h is required but these are not required so will remove it:
 #include "lte/protos/spgw_service.pb.h"
#include <folly/IPAddress.h>

File: lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.cpp (hunk 2 above)
Comment (ronit-kumar-acl) [reply lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.cpp]: Not required -> lte/protos/spgw_service.pb.h

File: lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.cpp (hunk 3 above)
Comment (ronit-kumar-acl) [reply lte/gateway/c/oai/tasks/grpc_service/AmfServiceImpl.cpp]: I will remove :  #include "spgw_service_handler.h" #include <folly/IPAddress.h>

**Comment by ronit-kumar-acl** [comment]
I have added Signed-off-by: ronit-kumar-acl <ronit.k@altencalsoftlabs.com> in the commit via GUI but still DCO check is failing any clue how to go about it in a GUI commit ? or a commit from a machine with -s is the only option ?

**Comment by gurrapualt** [comment]
This code pulled as part of #3366 request, so closing this.

**Comment by gurrapualt** [comment]
This code pulled as part of #3366 request, so closing this.

**Pull request closed without merging**